app/questionnaire/value_source_resolver.py

```diff
@@ -107,6 +114,30 @@ def _resolve_answer_value_source(
 
         return answer_value
 
+    def _resolve_progress_value_source(
+        self, value_source: Mapping
+    ) -> ValueSourceEscapedTypes | ValueSourceTypes | None:
+        if not self.progress_store:
+            raise NotImplementedError
+
+        identifier = value_source["identifier"]
+        selector = value_source["selector"]
+        if selector == "section":
+            return self.progress_store.get_section_status(identifier)
```

Comment:
Passing list item id from `self.location.list_item_id` if self.location is truthy else None

Reply:
We can't pass list item id without checking if section is repeating. However, having another look, I don't think we actually need this right now, since you can't reference a repeating section and since you can't reference the current section from itself, we shouldn't need the list item id.